tests/cacl/test_cacl_application.py

```diff
@@ -59,7 +63,128 @@ def docker_network(duthost):
 
     return docker_network
 
+@pytest.fixture(scope="function")
+def collect_ignored_rules(duthosts, rand_one_dut_hostname):
+    """
+    Collect existing iptables rules before test, set them as ignored as they are not related to CACL test cases.
+
+    Args:
+        duthosts: All DUTs belong to the testbed.
+        rand_one_dut_hostname: hostname of a random chosen dut to run test.
+
+    Returns:
+        None
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    # clear other control ACL rules before test to avoid miscalucation
+    duthost.command("acl-loader delete SNMP_ACL")
+    duthost.command("acl-loader delete NTP_ACL")
+    duthost.command("acl-loader delete SSH_ONLY")
+
+    logger.info('Waiting all rules to be cleaned')
+    wait_until(60, 2, 2, is_acl_rule_empty, duthost)
+    # acl-loader delete operation only deletes acl rules, it still has to wait some time for synchronizing iptables rules
+    if duthost.is_multi_asic:
+        time.sleep(60)
+    else:
+        time.sleep(10)
+
+    duthost = duthosts[rand_one_dut_hostname]
+    ignored_rules_v4 = duthost.command("iptables -S")["stdout_lines"]
+    ignored_rules_v6 = duthost.command("ip6tables -S")["stdout_lines"]
+
+    ignored_rules = {}
+    ignored_rules["v4"] = ignored_rules_v4
+    ignored_rules["v6"] = ignored_rules_v6
+    return ignored_rules
+
+@pytest.fixture(scope="function")
+def clean_scale_rules(duthosts, rand_one_dut_hostname, collect_ignored_rules):
+    """
+    Clear other control ACL rules before test to avoid miscalucation,
+    delete ACL template json file and clean ACL rules, recover configuration after test.
+
+    Args:
+        duthosts: All DUTs belong to the testbed.
+        rand_one_dut_hostname: hostname of a random chosen dut to run test.
+        collect_ignored_rules: ignored iptable/ip6table rules.
+
+    Returns:
+        None
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    yield
+
+    logger.info("delete tmp file and recover ACL configuration")
+    # delete the tmp file
+    duthost.file(path=SCALE_ACL_FILE, state='absent')
+    # recover ACL configuration
+    duthost.command("acl-loader delete SNMP_ACL")
+    duthost.command("acl-loader delete NTP_ACL")
+    duthost.command("acl-loader delete SSH_ONLY")
```

Comment:
Are you recovering? the commands are deleting.

Reply:
I clean all cacl rules before test then apply scale rules into db and compare rules. After test, I use these three commands to clean control acl rules and recover tb.

Reply:
As I discussed with Bing offline, just use  ```config reload``` to recover configuration, not call acl-loader delete cacl rules, in case of certain cacl rules must exist in the future.